### ISSUE TITLE
Teaching VCV rack on networked iMac's

### ISSUE THREAD

**Issue by oscarm77** [VCVRack/Rack#1617]
Hi,

I'm currently using vcvracks on a course I am teaching and it's excellent. It really makes student produce more imaginative creative work compared to other linear music making tools. 

One issue that is causing problems is that racks plugins are not hosted locally on machines. Therefore everytime students login all plugins are downloaded to a Host iMac which then sends them across the network. Racks stills works but takes along time to load plugins on start up and when adding a module. I was wondering if you know if anyone has encounter similar problems elsewhere or if there are plans to allow local storage of plugins alongside the web based account system?

Sorry for my somewhat laymens explanation. I hope this makes sense my knowledge of the subject is not great.

Thanks,
Oli

**Comment by AndrewBelt** [comment]
>racks plugins are not hosted locally on machines

Yes they are. Plugins are loaded from `<Rack user dir>/plugins`. Click "Library > Update all" and plugins will be downloaded to that location.

**Comment by z-plane** [comment]
It is not entirely clear how your machines are configured from your original post. In my experience edu environments push Active Directory as a way of dealing with security and authentication however, this tends to cause havoc with audio apps...

My current approach is to create a default edu account that students use to download the libraries for patches. There's no getting around the downloading of modules on machines and this step saves time so that they don't have to create their own account and add the right libraries during class. Hopefully, this encourages students to create a VCV account once they have got their heads the software basics and around broader concepts relating to modular workflows.

**Comment by oscarm77** [comment]
Hi,

Thanks for this, my explanation is not the best as I don't really understand how the network is configured but yes we have lots of problems running audio apps.

Are you saying that I can give a class the same login and still open racks on multiple machines? At the moment they have to download plugins to their user account which is hosted on the network rather than on the iMac they are working on. Obviously if the whole class is doing this at the same time we run into problems.


________________________________

**Comment by AndrewBelt** [comment]
There are two options I can think of.
- Copy a "master" Rack user dir to students' Documents folders.
- Run Rack with [command line arguments](https://vcvrack.com/manual/Installing.html#command-line-usage) (using a shortcut you create on students computers) with something like `Rack.exe -u /path/to/network/storage/Rack`
